situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/processor/oteltrace/RawTrace.java

```diff
@@ -0,0 +1,141 @@
+package com.amazon.situp.plugins.processor.oteltrace;
+
+import io.opentelemetry.proto.resource.v1.Resource;
+import io.opentelemetry.proto.trace.v1.Span;
+import io.opentelemetry.proto.common.v1.InstrumentationLibrary;
+import org.apache.commons.codec.binary.Hex;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.Instant;
+import java.util.Map;
+
+public class RawTrace {
+
+    private static final BigDecimal MILLIS_TO_NANOS = new BigDecimal(1_000_000);
+    private static final BigDecimal SEC_TO_MILLIS = new BigDecimal(1_000);
+
+    private final String traceID;
+    private final String spanID;
+    private final String parentSpanID;
+    private final String name;
+    private final String startTime;
+    private final String endTime;
+    private final String instrumentationName;
+    private final String instrumentationVersion;
+    private final String serviceName;
+    private final Map<String, String> resourceAttributes;
+
+
+    private RawTrace(RawTraceBuilder builder) {
+        this.traceID = builder.traceID;
+        this.spanID = builder.spanID;
+        this.parentSpanID = builder.parentSpanID;
+        this.name = builder.name;
+        this.startTime = builder.startTime;
+        this.endTime = builder.endTime;
+        this.instrumentationName = builder.instrumentationName;
+        this.instrumentationVersion = builder.instrumentationVersion;
+        this.serviceName = builder.serviceName;
+        this.resourceAttributes = builder.resourceAttributes;
+    }
+
+    /**
+     * Builder class
+     **/
+    public static class RawTraceBuilder {
+        private String traceID;
+        private String spanID;
+        private String parentSpanID;
+        private String name;
+        private String startTime;
+        private String endTime;
+        private String instrumentationName;
+        private String instrumentationVersion;
+        private String serviceName;
+        private Map<String, String> resourceAttributes;
+
+        public RawTraceBuilder() {
+        }
+
+        public RawTraceBuilder setTraceID(String traceID) {
+            this.traceID = traceID;
```

Comment:
as suggested by @kowshikn go ahead and update the PR to include Preconditions.check for these